File: example/src/main/java/ru/virgil/example/user/UserDetails.java
package ru.virgil.example.user;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import ru.virgil.utils.image.IBaseEntity;
import ru.virgil.example.security.SecurityUserDetails;
import ru.virgil.example.system.IdentifiedEntity;

import javax.persistence.OneToOne;

@javax.persistence.Entity
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@Data
public class UserDetails extends IdentifiedEntity implements IBaseEntity {

    @OneToOne
    private SecurityUserDetails securityUserDetails;
    private String name;

}

File: example/src/main/java/ru/virgil/example/user/UserDetails.kt
package ru.virgil.example.user

import ru.virgil.example.security.SecurityUserDetails
import ru.virgil.example.system.entity.IdentifiedEntity
import ru.virgil.utils.base.entity.Identified
import ru.virgil.utils.base.entity.Timed
import javax.persistence.Entity
import javax.persistence.OneToOne

@Entity
class UserDetails(
    @OneToOne
    private val securityUserDetails: SecurityUserDetails,
    var name: String? = null,
) : IdentifiedEntity(), Identified, Timed
